pkg/apis/app/v1alpha1/kogitoinfra_types.go

```diff
@@ -19,6 +19,19 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// Resource provide reference infra resource
+type Resource struct {
+
+	// Kind describe the kind of referred resource for example, app.infinispan.org/v1
+	Kind string `json:"kind,omitempty"`
+
+	// Namespace where referred resource exists
+	Namespace string `json:"namespace,omitempty"`
+
+	// Name of referred resource
+	Name string `json:"name,omitempty"`
+}
```

Comment:
@vaibhavjainwiz I believe we would need the `apiVersion` attribute here as well, with is the `app.infinispan.org/v1`. So the interface should be something like:

```suggestion
// Resource provide reference infra resource
type Resource struct {
	 // ApiVersion describes the API Version of referred Kubernetes resource for example, app.infinispan.org/v1
   	ApiVersion string `json:"apiVersion"`

	// Kind describes the kind of referred Kubernetes resource for example, Infinispan
	Kind string `json:"kind"`

	// Namespace where referred resource exists, default to the current namespace
	Namespace string `json:"namespace,omitempty"`

	// Name of referred resource
	Name string `json:"name"`
}
```

Also I've removed the `omitempty` attribute from the required properties.

Reply:
+1

Reply:
done